### PR TITLE
Pinned message indicator

### PR DESCRIPTION
**Pull request by erinb90**
PR for issue #19 (epic #18)

### CODE REVIEW

File: test/androidTest/java/org/thoughtcrime/securesms/espresso/BaseRecyclerHelper.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    public T assertNoIdAt(int id, int position, boolean checkVisible) {
+        try {
+            onView(withId(android.R.id.list))
+                .perform(scrollToPosition(position))
+                .check(matches(atPosition(position, hasDescendant(withId(id)), id)));
+
+            throw new Error("Helper.assertNoIdAt: View found matching \"" + id + "\" at child #" + position);
+        } catch (Error e) {
```
Comment (g-harel) [inline test/androidTest/java/org/thoughtcrime/securesms/espresso/BaseRecyclerHelper.java]: I want to make this more specific

File: test/androidTest/java/org/thoughtcrime/securesms/espresso/BaseRecyclerHelper.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.thoughtcrime.securesms.espresso.ViewMatchers.atPosition;
+
+class BaseRecyclerHelper<T> extends Helper {
+
+    /* ASSERTIONS */
+
+    public T assertIdAt(int id, int position, boolean checkVisible) {
```
Comment (g-harel) [inline test/androidTest/java/org/thoughtcrime/securesms/espresso/BaseRecyclerHelper.java]: I want to split id and visible assertions.

**Review by g-harel — CHANGES_REQUESTED** [review]
the only changes are about my test code :+1:

File: test/androidTest/java/org/thoughtcrime/securesms/espresso/BaseRecyclerHelper.java (hunk 2 above)
Comment (erinb90) [reply test/androidTest/java/org/thoughtcrime/securesms/espresso/BaseRecyclerHelper.java]: Oh ok cool I wasn't sure if you were finished the tests or not

Comment (g-harel) [reply test/androidTest/java/org/thoughtcrime/securesms/espresso/BaseRecyclerHelper.java]: they are functionally finished, but I wanna make em nicer. was getting frustrated with espresso at the end of it ...

File: res/layout/conversation_item_received.xml [hunk 3]
```diff
@@ -105,6 +120,8 @@
                     app:scaleEmojis="true"
                     tools:text="boop"/>
```
Comment (veken1199) [inline res/layout/conversation_item_received.xml:122]: Extra space is not needed

File: src/org/thoughtcrime/securesms/ConversationFragment.java [hunk 4]
```diff
@@ -248,10 +248,7 @@ private void setCorrectMenuVisibility(Menu menu) {
   }
 
   private void setCorrectPinVisibility(Menu menu, MessageRecord messageRecord, boolean actionMessage) {
-    PinnedMessagesHandler pinHandler = new PinnedMessagesHandler(getContext());
-    MessagingDatabase databaseToQuery = pinHandler.getAppropriateDatabase(messageRecord);
-
-    if (databaseToQuery.isPinned(messageRecord.getId())) {
+    if (messageRecord.isPinned()) {
```
Comment (veken1199) [inline src/org/thoughtcrime/securesms/ConversationFragment.java:251]: This is so much better, Will take a look at the actual implementation

File: test/unitTest/java/org/thoughtcrime/securesms/pinmessages/PinnedMessagesHandlerTests.java [hunk 5]
```diff
@@ -47,6 +47,7 @@ public void testHandlerPinMessagesSmsSuccess() {
         boolean result = handler.handlePinMessage(messageRecordSms, smsDatabase);
 
         verify(smsDatabase).pinMessage(1);
+        verify(smsDatabase).markAsPinned(1);
```
Comment (veken1199) [inline test/unitTest/java/org/thoughtcrime/securesms/pinmessages/PinnedMessagesHandlerTests.java:50]: I was going to mention this, to test the markAsPinned() and markAsUnPinned()

**Review by g-harel — APPROVED** [review]

**Review by veken1199 — APPROVED** [review]
Very small formatting issues, However, it is acceptable

**Comment by veken1199** [comment]
@erinb90 can you add the story # and the estimated hours ?

File: res/layout/conversation_item_received.xml (hunk 3 above)
Comment (erinb90) [reply res/layout/conversation_item_received.xml:122]: They usually have a space before a closing xml tag

**Comment by erinb90** [comment]
@veken1199 do you mean the actual implementation time? I was planning to do that only after the pr is merged in case I need to change something.

**Comment by erinb90** [comment]
Pinned indicator looks like this:
screenshot_2018-03-01-01-03-37

**Pull request merged**